Source/WebKit/Shared/Cocoa/ArgumentCodersCocoa.h

```
@@ -246,12 +246,21 @@ template<typename T> struct ArgumentCoder<RetainPtr<T>> {
     template<typename U = T, typename = IsObjCObject<U>>
     static void encode(Encoder& encoder, const RetainPtr<U>& object)
     {
+        if (!object) {
+            encoder << false;
+            return;
+        }
+
+        encoder << true;
         ArgumentCoder<U *>::encode(encoder, object.get());
     }
 
     template<typename U = T, typename = IsObjCObject<U>>
     static std::optional<RetainPtr<U>> decode(Decoder& decoder)
     {
+        auto isEngaged = decoder.template decode<bool>();
+        if (!isEngaged.value_or(true))
```

If decoding the bool fails, we should return nullopt instead of nullptr.